packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -1280,6 +1272,15 @@ function simplifyStruct<T>(typeDef: StructType, context: SimplifyContext<T>): Si
 		}
 	}
 	for (const field of typeDef.fields) {
+		let keep = true
+		handleAttributes(field.attributes, context.options.context, (handler, config) => {
+			if (keep && handler.filterElement?.(config, context.options.context) === false) {
```

Comment:
does `filterElement` not for sure return `false`? i guess it can be `undefined` here but it's not immediately clear to me why `undefined` wouldn't also make `keep = false`

not that this logic is new to this PR

Reply:
`filterElement` always returns `true` or `false`, but it will be `undefined` if `handler.filterElement` is undefined. Attributes that don't have this handler method shouldn't influence whether a field is kept or not.

Reply:
I have plans to refactor `handleAttributes` slightly so that it takes a method string (`'filterElement` in this case) and only loops over the attributes that have this method. Both to improve performance and to make things slighly easier to read.